growattServer/__init__.py

```diff
@@ -610,3 +613,34 @@ def update_mix_inverter_setting(self, serial_number, setting_type, parameters):
         response = self.session.post(self.get_url('newTcpsetAPI.do'), params=settings_params)
         data = json.loads(response.content.decode('utf-8'))
         return data
+
+    def update_ac_inverter_setting(self, serial_number, setting_type, parameters):
```

Comment:
This function appears to just be a copy/paste of the same settings function for the mix_inverter - we should have a common function for this (used internally within the library) which is called by these two function with just the parts that change (the target URL and the default parameters).

Reply:
If you're unsure what I mean here let me know and I'll give you a code snippet.

Reply:
This is a good suggestion.

I'm thinking this could be implemented through a minor change to the function definition, like `update_ac_inverter_setting(self, serial_number, setting_type, parameters, mix = True):`

Reply:
I see how you've chosen to implement this, however, what if another type comes along? Suddenly the boolean needs to be a string that needs to be parsed etc. 
My suggestion above provides more future proofing, yes there's a few more lines of code for each function but at least each on is distinct and offers scalability for more similar functions in the future.

Reply:
Sure. I take your point. The parameters seem to vary depending not only on "mix" vs. "AC-coupled" but on the model type. I'll put it together the way you suggest this evening.

I'm working on the get data for SPA-type inverters. I suspect I'll apply a similar approach.